src/binary_heap.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals, division
+
+
+class BinaryHeap(object):
+    """Binary Heap data structure class object"""
+    def __init__(self, iterable=None):
+        """Init an instance with the option to pass in an iterable"""
+        self._list = []
+        try:
+            for element in iterable:
+                self._list.append(element)
+            self._list.sort()
```

Comment:
Your heap should be able to work without using the builtin sorting functionality of the Python list.

You've already implemented an algorithm for heap-ifying your list in both the `push` and the `pull` functions. Why not re-use this functionality on initialization? Consider abstracting the algorithm into a function which can be used by `push`, `pull` and `__init__`